src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HealthHeartbeatProvider.cs

```diff
@@ -0,0 +1,226 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    /// <summary>
+    /// Implementation of health heartbeat functionality.
+    /// </summary>
+    internal class HealthHeartbeatProvider : IDisposable, IHeartbeatProvider
+    {
+        /// <summary>
+        /// The default interval between heartbeats if not specified by the user
+        /// </summary>
+        public static int DefaultHeartbeatIntervalMs = 5000;
+
+        /// <summary>
+        /// The default fields to include in every heartbeat sent. Note that setting the value to '*' includes all default fields.
+        /// </summary>
+        public static string DefaultAllowedFieldsInHeartbeatPayload = "*";
```

Comment:
Can it be a `private const string` ?

Reply:
Changes to the design (blacklist instead of my original whitelist) makes this go away. Again - public for testing purposes...